### PR TITLE
E2E: Fee middleware counterparty payee not set

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #1686

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: e2e/fee_middleware_test.go [hunk 1]
```diff
@@ -429,6 +429,108 @@ func (s *FeeMiddlewareTestSuite) TestMsgPayPacketFee_SingleSender_TimesOut() {
...
+			s.Require().Len(packets, 1)
+			actualFee := packets[0].PacketFees[0].Fee
+
+			s.Require().True(actualFee.RecvFee.IsEqual(testFee.RecvFee))
+			s.Require().True(actualFee.AckFee.IsEqual(testFee.AckFee))
+			s.Require().True(actualFee.TimeoutFee.IsEqual(testFee.TimeoutFee))
+		})
+	})
+
+	t.Run("balance should be lowered by sum of recv ack and timeout", func(t *testing.T) {
```
Comment (crodriguezvega) [inline e2e/fee_middleware_test.go:503]:
```suggestion
	t.Run("balance should be lowered by sum of recv, ack and timeout", func(t *testing.T) {
```

File: e2e/fee_middleware_test.go [hunk 2]
```diff
@@ -429,6 +429,108 @@ func (s *FeeMiddlewareTestSuite) TestMsgPayPacketFee_SingleSender_TimesOut() {
...
+		t.Run("no incentivized packets", func(t *testing.T) {
+			packets, err := s.QueryIncentivizedPacketsForChannel(ctx, chainA, channelA.PortID, channelA.ChannelID)
+			s.Require().NoError(err)
+			s.Require().Empty(packets)
+		})
+
+		packetId := channeltypes.NewPacketId(channelA.PortID, channelA.ChannelID, 1)
+		packetFee := feetypes.NewPacketFee(testFee, chainAWallet.Bech32Address(chainA.Config().Bech32Prefix), nil)
+
+		t.Run("should succeed", func(t *testing.T) {
```
Comment (crodriguezvega) [inline e2e/fee_middleware_test.go:484]: Maybe something extra in the description to make it more explicit? Like `pay packet fee should succeed`?

**Review by crodriguezvega — APPROVED** [review]
Nice, @chatton!

File: e2e/fee_middleware_test.go (hunk 2 above)
Comment (chatton) [reply e2e/fee_middleware_test.go:484]: based on the testing, the resulting message will look like `TestPayPacketFeeAsync_SingleSender_NoCounterPartyAddress/pay_packet_fee/should_succeed`

File: e2e/fee_middleware_test.go [hunk 3]
```diff
@@ -426,6 +426,108 @@ func (s *FeeMiddlewareTestSuite) TestMsgPayPacketFee_SingleSender_TimesOut() {
...
+	})
+
+	t.Run("pay packet fee", func(t *testing.T) {
+		t.Run("no incentivized packets", func(t *testing.T) {
+			packets, err := s.QueryIncentivizedPacketsForChannel(ctx, chainA, channelA.PortID, channelA.ChannelID)
+			s.Require().NoError(err)
+			s.Require().Empty(packets)
+		})
+
+		packetId := channeltypes.NewPacketId(channelA.PortID, channelA.ChannelID, 1)
```
Comment (colin-axner) [inline e2e/fee_middleware_test.go:478]:
```suggestion
		packetId := channeltypes.NewPacketId(chainATx.Packet.GetSourcePort(), chainATx.Packet.GetSourceChannel(), chainATx.Packet.GetSequence())
```

I think at least modifying the sequence to use the returned chainATx.Packet.GetSequence makes sense, I have no preference on the other arguments, I just figure this code will be copied a lot so might make more sense to derive the values

File: e2e/fee_middleware_test.go [hunk 4]
```diff
@@ -426,6 +426,108 @@ func (s *FeeMiddlewareTestSuite) TestMsgPayPacketFee_SingleSender_TimesOut() {
...
+		packetFee := feetypes.NewPacketFee(testFee, chainAWallet.Bech32Address(chainA.Config().Bech32Prefix), nil)
+
+		t.Run("should succeed", func(t *testing.T) {
+			var err error
+			payPacketFeeTxResp, err = s.PayPacketFeeAsync(ctx, chainA, chainAWallet, packetId, packetFee)
+			s.Require().NoError(err)
+			s.AssertValidTxResponse(payPacketFeeTxResp)
```
Comment (colin-axner) [inline e2e/fee_middleware_test.go:485]: love this helper function :heart:

**Review by colin-axner — APPROVED** [review]
Great work! Looks super clean :)

File: e2e/fee_middleware_test.go (hunk 3 above)
Comment (chatton) [reply e2e/fee_middleware_test.go:478]: ah definitely, good catch.

**Review by seantking — APPROVED** [review]
<3

**Pull request merged**